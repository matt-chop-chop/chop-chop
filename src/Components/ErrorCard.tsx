import { Button, Flex, Link, Text } from "@chakra-ui/react";
import { InfoOutlineIcon } from "@chakra-ui/icons";
import NextLink from "next/link";
import { useRouter } from "next/router";

type ErrorCardProps = {
  error: Error;
};

const ErrorCard = ({ error }: ErrorCardProps) => {
  const router = useRouter();
  const { pathname } = router;
  const id = router.query["id"];

  return (
    <Flex
      alignItems="center"
      background="var(--chakra-colors-light-background)"
      border="2px solid var(--chakra-colors-light-emphasis)"
      borderRadius={4}
      flexDirection="column"
      justifyContent="center"
      minHeight="250px"
      p={4}
      width="100%"
    >
      <Text textAlign="center" variant="body-bold">
        <InfoOutlineIcon mr={2} />
        This recipe could not be loaded due to the following error: &quot;
        {error.message}&quot;
      </Text>
      <Text mt={2} variant="body">
        Try refreshing the page
      </Text>
      <Link as={NextLink} href={pathname === "/" ? "/" : `${id}`} mt={2}>
        <Button variant="primary">Refresh</Button>
      </Link>
    </Flex>
  );
};

export default ErrorCard;
