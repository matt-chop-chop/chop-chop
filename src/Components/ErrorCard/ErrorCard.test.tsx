import { shallow } from "enzyme";
import React from "react";
import ErrorCard, { ErrorCardProps } from "./ErrorCard";
import { useRouter } from "next/router";

jest.mock("next/router", () => ({
  __esModule: true,
  useRouter: jest.fn().mockReturnValue({ pathname: "/", query: { id: "123" } }),
}));

describe("ErrorCard", () => {
  const props: ErrorCardProps = {
    error: new Error("There has been an error."),
    showBorder: false,
    mt: "12px",
  };

  it("is exported from ErrorCard", () => {
    expect(ErrorCard).toBeDefined();
  });

  it("renders as expected", () => {
    const wrapper = shallow(<ErrorCard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders with default props", () => {
    const wrapper = shallow(<ErrorCard error={props.error} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders a link to page relating to the id in the url if pathname is not /", () => {
    const wrapper = shallow(<ErrorCard error={props.error} />);

    (useRouter as jest.Mock).mockReturnValue({
      pathname: "test",
      query: {
        id: "123",
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
