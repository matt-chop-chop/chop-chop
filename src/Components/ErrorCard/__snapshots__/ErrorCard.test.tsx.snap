// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorCard renders as expected 1`] = `
<Flex
  alignItems="center"
  background="var(--chakra-colors-light-background)"
  border=""
  borderRadius={4}
  flexDirection="column"
  justifyContent="center"
  minHeight="250px"
  mt="12px"
  p={4}
  width="100%"
>
  <Text
    textAlign="center"
    variant="body-bold"
  >
    <InfoOutlineIcon
      mr={2}
    />
    This recipe could not be loaded due to the following error: "
    There has been an error.
    "
  </Text>
  <Text
    mt={2}
    variant="body"
  >
    Try refreshing the page
  </Text>
  <Link
    _hover={
      {
        "transform": "scale(1.1)",
        "transition": "transform 250ms ease-in",
      }
    }
    as={
      {
        "$$typeof": Symbol(react.forward_ref),
        "default": [Circular],
        "render": [Function],
      }
    }
    href="/"
    mt={2}
    transition="transform 250ms ease-out"
  >
    <Button
      variant="primary"
    >
      Refresh
    </Button>
  </Link>
</Flex>
`;

exports[`ErrorCard renders with default props 1`] = `
<Flex
  alignItems="center"
  background="var(--chakra-colors-light-background)"
  border="2px solid var(--chakra-colors-light-text)"
  borderRadius={4}
  flexDirection="column"
  justifyContent="center"
  minHeight="250px"
  mt="0px"
  p={4}
  width="100%"
>
  <Text
    textAlign="center"
    variant="body-bold"
  >
    <InfoOutlineIcon
      mr={2}
    />
    This recipe could not be loaded due to the following error: "
    There has been an error.
    "
  </Text>
  <Text
    mt={2}
    variant="body"
  >
    Try refreshing the page
  </Text>
  <Link
    _hover={
      {
        "transform": "scale(1.1)",
        "transition": "transform 250ms ease-in",
      }
    }
    as={
      {
        "$$typeof": Symbol(react.forward_ref),
        "default": [Circular],
        "render": [Function],
      }
    }
    href="/"
    mt={2}
    transition="transform 250ms ease-out"
  >
    <Button
      variant="primary"
    >
      Refresh
    </Button>
  </Link>
</Flex>
`;
